File: src/main/java/me/muksc/tacztweaks/mixin/accessor/JsonFileCodecConfigAccessor.java
package me.muksc.tacztweaks.mixin.accessor;

import com.google.gson.Gson;
import dev.isxander.yacl3.config.v3.JsonFileCodecConfig;
import org.spongepowered.asm.mixin.Mixin;
import org.spongepowered.asm.mixin.gen.Accessor;

import java.nio.file.Path;

@Mixin(value = JsonFileCodecConfig.class, remap = false)
public interface JsonFileCodecConfigAccessor {
    @Accessor("configPath")
    Path tacztweaks$getConfigPath();

    @Accessor
    Gson getGson();
}

File: src/main/kotlin/me/muksc/tacztweaks/mixininterop/JsonFileCodecConfigAccessor.kt
package me.muksc.tacztweaks.mixininterop

import me.muksc.tacztweaks.mixin.accessor.JsonFileCodecConfigAccessor
import java.nio.file.Path

inline val JsonFileCodecConfigAccessor.configPath: Path
    get() = `tacztweaks$getConfigPath`()